### PR TITLE
KT-31126 JS. Fix super constructor call. Old backend

### PR DESCRIPTION
**Pull request by turansky**
Primary ordinary -> secondary external

### CODE REVIEW

**Comment by romanart** [comment]
Thank you for PR. Could please add a test which your change fixes?

**Comment by turansky** [comment]
Test for constructor require external `js` resource (as in my [test example](https://github.com/turansky/kotlin-battleground/tree/master/KT-31126))
* [Test](https://github.com/turansky/kotlin-battleground/blob/master/KT-31126/src/test/kotlin/com/test/DataProviderTest.kt)
* [External resource](https://github.com/turansky/kotlin-battleground/blob/master/KT-31126/src/main/resources/data-provider.js)

I don't see tests with external in `js.translator`.
Is it possible to use external resources in `js.translator`?
Is there better place for such tests?

**Comment by romanart** [comment]
Hope this could help you
https://github.com/JetBrains/kotlin/blob/master/js/js.translator/testData/box/jsModule/externalClass.kt
https://github.com/JetBrains/kotlin/blob/master/js/js.translator/testData/box/jsModule/externalClass.js

Think `js/js.translator/testData/box/jsModule` is a good place for such test.

**Comment by turansky** [comment]
> Could please add a test which your change fixes?
Done

**Comment by romanart** [comment]
Pushed to CI. Once it done without failures will merge your fix.

Thank you :rocket:

**Comment by turansky** [comment]
@romanart What about one more review #3219? Tests already added.

**Comment by romanart** [comment]
@turansky Could you please add the following lines in the beginning of `externalConstructor.kt`?
```
// EXPECTED_REACHABLE_NODES: 1334
// MODULE_KIND: AMD
```
Those lines says test infra how to run this test. Otherwise it fails with 
```
CALL_TO_JS_MODULE_WITHOUT_MODULE_SYSTEM: Can't access constructor of 'A' marked with @JsModule annotation from non-modular project (11,25)
```

**Comment by turansky** [comment]
@romanart Which task generate following code:
```
// EXPECTED_REACHABLE_NODES: 1334
```

**Comment by romanart** [comment]
`gradle :js:js.tests:test --tests "org.jetbrains.kotlin.js.test.ir.semantics.IrBoxJsTestGenerated$JsModule.testExternalClass"`

**Comment by turansky** [comment]
@romanart comments added :)

File: js/js.translator/testData/box/jsModule/externalConstructor.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// EXPECTED_REACHABLE_NODES: 1334
+// MODULE_KIND: AMD
+package foo
+
+@JsModule("lib")
```
Comment (bashor) [inline js/js.translator/testData/box/jsModule/externalConstructor.kt:5]: Isn't it reproduced without any module systems?

**Comment by bashor** [comment]
It would be nice to check that we have tests for all cases? You can just create an issue for that.

|ctor|super ctor| super is external |
| :--- | :--- | :--- | 
|primary|primary| false |
|primary|secondary| false |
|secondary|primary| false |
|secondary|secondary| false |
|primary|primary| true |
|primary|secondary| true |
|secondary|primary| true |
|secondary|secondary| true |

cc @romanart

File: js/js.translator/testData/box/jsModule/externalConstructor.kt (hunk 1 above)
Comment (turansky) [reply js/js.translator/testData/box/jsModule/externalConstructor.kt:5]: Works in both cases.
Modular variant can be simply tested in [separate project](https://github.com/turansky/kotlin-battleground/blob/master/KT-31126/src/test/kotlin/foo/externalConstructor.kt)

**Comment by turansky** [comment]
> You can just create an issue for that.

@bashor You = @me?

File: js/js.translator/testData/box/jsModule/externalConstructor.kt (hunk 1 above)
Comment (bashor) [reply js/js.translator/testData/box/jsModule/externalConstructor.kt:5]: My thought was that (without the fix) it should be reproducible without any module system. Isn't it?

**Comment by bashor** [comment]
@turansky, yes, could you please create an issue for https://github.com/JetBrains/kotlin/pull/3265#issuecomment-609901039

File: js/js.translator/testData/box/jsModule/externalConstructor.kt (hunk 1 above)
Comment (turansky) [reply js/js.translator/testData/box/jsModule/externalConstructor.kt:5]: Yes, it is.

Comment (turansky) [reply js/js.translator/testData/box/jsModule/externalConstructor.kt:5]: Modular system used like more typical.

**Comment by turansky** [comment]
[KT-38114](https://youtrack.jetbrains.com/issue/KT-38114)

**Pull request merged**